### PR TITLE
Adds Configurable GPU Support

### PR DESCRIPTION
**Pull request by justinthelaw**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -12,6 +17,7 @@ class Model:
     llm = Llama(
         model_path=backend_config.model.source,
         n_ctx=backend_config.max_context_length,
+        n_gpu_layers=-1 if GPU_ENABLED == True else 0,
```
Comment (gerred) [inline main.py:20]: In a next PR, let's make the n_gpu_layers configurable. I've had wonky issues if I'm near limits on VRAM where it'll still offload all of the layers and _technically_ work but I get garbage back during inferencing vs. just a hard failure.

File: pyproject.toml [hunk 2]
```diff
@@ -6,11 +6,22 @@ description = "llama-cpp-py based backend for LeapfrogAI"
 authors = [{ name = "LeapfrogAI Authors", email = "ai@defenseunicorns.com" }]
 license = { file = "LICENSE" }
 readme = "README.md"
-requires-python = ">=3.10, <3.12"
-dependencies = ["llama-cpp-python >= 0.2.20", "leapfrogai == 0.4.0rc1"]
+requires-python = ">=3.11.4"
```
Comment (gerred) [inline pyproject.toml:9]: keep <3.12 in there, 3.12 still breaks things and Torch is on a very delayed update schedule.

**Review by gerred — CHANGES_REQUESTED** [review]
Only change I'd explicitly request is pinning under 3.12, because of FastAPI/Pydantic/Torch issues.

File: main.py (hunk 1 above)
Comment (CollectiveUnicorn) [reply main.py:20]: Added an issue for this, https://github.com/defenseunicorns/leapfrogai-backend-llama-cpp-python/issues/10

**Review by CollectiveUnicorn — APPROVED** [review]

**Pull request merged**